### ISSUE TITLE
Add data aws_ebs_volumes

### ISSUE THREAD

**Issue by flmmartins** [terraform-providers/terraform-provider-aws#12243]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We have the data aws_ebs_volume where we can fetch one volume.

### New or Affected Resource(s)

We would create something like:

```

data aws_ebs_volumes my_volumes {
    filter {
       name = "tag:cluster_volumes"
       values = ["cluster x"]
    }
}

```

This would return **multiple ebs volumes**. The behaviour would be similar data aws_instances.

**Comment by mzraly** [comment]
This enhancement would be helpful implementing code that finds or creates a persistent EBS volume, e.g. for a database instance.  Ideally we'd be able to search for the most recently created EBS volume matching a set of volume filters, then if we don't find one (if the list of matching volumes is empty) create a volume with the desired characteristics, then reference either the found volume id or the created one in subsequent code.

**Comment by bflad** [comment]
A new `aws_ebs_volumes` data source has been merged and will release with version 2.67.0 of the Terraform AWS Provider, later this week. Thanks to @jen20 for the implementation. 👍
